infrastructure/lib/constructs/lambda-fn.ts

```diff
@@ -0,0 +1,220 @@
+   */
+  readonly handlerPath?: string;
+
+  /**
+   * Any additional props to apply to the NodeJsFunction.
+   *
+   * Additional prop values specified here will override the defaults, including
+   * the `entry` and `environment` if one is provided in this struct. Be careful.
+   */
+  readonly functionPropsOverride?: lambdaNodeJs.NodejsFunctionProps;
+
+  /**
+   * The VPC where resources should be created
+   */
+  readonly lambdaVpc?: ec2.IVpc;
+
+  /**
+   * A prop to determine if the underlying Lambda function has read permissions
+   * for secrets managers.
+   */
+  readonly smtpSecrets?: secretsmanager.ISecret;
+
+  /**
+   * The DynamoDB table where data is stored.
+   */
+  readonly table?: dynamodb.ITable;
+
+  /**
+   * Optional table permissions to grant to the Lambda function.
+   * One of the following may be specified: "ALL", "READ", "READ_WRITE", "WRITE".
+   *
+   * @default - Read/write access is given to the Lambda function if no value is specified.
+   */
+  readonly tablePermissions?: string;
+
+  /**
+   * Secure bucket for S3 functions
+   */
+  readonly bucket?: SecureBucket;
+
+  /**
+   * Optional bucket permissions to grant to the Lambda function.
+   * One of the following may be specified: "READ", "PUT", "DELETE".
+   *
+   * @default - Read access is given to the Lambda function if no value is specified.
+   */
+  readonly bucketPermissions?: string;
+
+  /**
+   * The SQS queue where messages are sent
+   */
+  readonly queue?: sqs.IQueue;
+
+  /**
+   * Optional bucket permissions to grant to the Lambda function.
+   * One of the following may be specified: "SEND", "CONSUME".
+   *
+   * @default - Read access is given to the Lambda function if no value is specified.
+   */
+  readonly queuePermissions?: string;
+
+  /**
+   * Optional param to create an SQS event source to receive queue messages
+   */
+  readonly createEventSource?: boolean;
+
+  /**
+   * Optional param to change the batch size of messages received by the lambda
+   * from the SQS event source
+   */
+  readonly batchSize?: number;
+
+  /**
+   * The State Machine that will start execution when this function
+   * is called.
+   */
+  readonly stateMachine?: sfn.IStateMachine;
+}
+
+export class ApiFunctiontest extends cdk.Construct {
+  /**
+   * Required resources
+   * @param fn - the NodejsFunction, named with the @id passed to the function
+   */
+  public readonly fn: lambda.Function;
+  /**
+   * Optional resources to add to function:
+   * @param method - HTTP Method, indicates that the function uses API Gateway
+   * @param table - DynamoDB table connection
+   * @param bucket - S3 bucket connection
+   * @param queue - SQS queue connection
+   * @param stateMachine - State Machine connection
+   */
+  public readonly method: HttpMethod;
+  public readonly table: dynamodb.ITable;
+  public readonly bucket: s3.IBucket;
+  public readonly queue: sqs.IQueue;
+  public readonly stateMachine?: sfn.IStateMachine;
```

Comment:
Do we needed the `?` for `stateMachine` only? Just wondering.

Reply:
Good catch, forgot to remove the `?`.

Reply:
Was removed in 1145be5180547470cc60ea395e8fe493755a7acc